### PR TITLE
69: Added project start and end statistics

### PR DESCRIPTION
**Pull request by eremozdemir**
## 📝 Description

> Please include a summary of the change and which issue is fixed.  
> Include relevant motivation and context.  
> List any dependencies that are required for this change.

### ProjectReport.init Implementation
This method implements the initialization logic for the `ProjectReport` class, which aggregates file-level statistics into project-level metrics.

#### What it does:

- Takes a list of `FileReport` objects as input
- Extracts creation and modification dates from each file report
- Calculates two key project statistics:
- Project Start Date: The earliest file creation date across all files
- Project End Date: The latest file modification date across all files
- Handles edge cases where files may have missing date information
- Creates a `StatisticIndex`with the calculated project-level statistics
Initializes the parent `BaseReport`class with the aggregated statistics

#### Key features:

- Filters out None values to handle files with missing date metadata
- Only creates statistics when valid dates are available
- Uses `min()` and `max()` functions to efficiently find earliest/latest dates
- Follows the established pattern of wrapping calculated values in `Statistic`objects

#### Testing: Created comprehensive unit tests in `test_project_report.py` that cover:
- Basic functionality with multiple files
- Edge cases (empty file lists, single files, missing dates)
- Complex scenarios with multiple files and various date patterns
- Negative assertions using assertFalse to verify incorrect assumptions
- Inheritance testing to ensure `BaseReport` methods work properly

**Closes:**  #69 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [x] ⚡ Performance improvement

---

## 🧪 Testing

> Please describe how you tested this PR (both manually and with tests).  
> Provide instructions so we can reproduce.

- [x] Test A:  `test_project_report.py` class
- [ ] Test B

---

## ✓ Checklist

- [ ] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [x] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

</details>

### CODE REVIEW

File: src/classes/report.py [hunk 1]
```diff
@@ -48,20 +48,50 @@ def get_filename(self):
...
+            end_date = max(date_modified_list)              # Find the maximum (latest) date from all modification dates
+            project_end_stat = Statistic(ProjectStatCollection.PROJECT_END_DATE.value, end_date)
+
+        # Build list of project-level statistics
+        # Only add statistics that were successfully calculated
+        project_stats = []
+        
+        # Add start date statistic if we found any creation dates
+        if date_created_list:
+            project_stats.append(project_start_stat)
```
Comment (sjsikora) [inline src/classes/report.py]: nit: This is so minor, but for readability, we can move this line up into the earlier if statement?

So instead of

```python
if date_created_list:
   start_date = min(date_created_list)
   project_start_stat = Statistic(ProjectStatCollection.PROJECT_START_DATE.value, start_date)
          
...
if date_created_list:
   project_stats.append(project_start_stat)   
```
we would have :
```python
if date_created_list:
   start_date = min(date_created_list)
   project_start_stat = Statistic(ProjectStatCollection.PROJECT_START_DATE.value, start_date)
   project_stats.append(project_start_stat) 
```

And the same with "if date_modified_list"?

**Review by jademola — CHANGES_REQUESTED** [review]
Report section looks good, however due to changes since the PR, conflicts now exist in the import section. Please resolve, then we are good to go. Also see Sam's comment

**Comment by eremozdemir** [comment]
@sjsikora I had to do some decent conflict resolution for the project class to incoporate both init functions, would you mind reviewing these changes and/ or anyone else that has worked on the ProjectReport class
